fenixedu-ist-cms-components/src/main/java/pt/ist/fenixedu/cmscomponents/domain/unit/components/UnitCourses.java

```diff
@@ -104,4 +114,47 @@ public DepartmentUnit getPersonDepartmentUnit() {
         final Department department = employee == null ? null : employee.getCurrentDepartmentWorkingPlace();
         return department == null ? null : department.getDepartmentUnit();
     }
+
+    public List<ExecutionYear> getValidExecutionYearsForUnit(Unit unit) {
+        List<ScientificAreaUnit> scientificAreaUnits = unit.getSubUnits().stream()
+                .filter(Unit::isScientificAreaUnit)
+                .map(ScientificAreaUnit.class::cast)
+                .sorted(ScientificAreaUnit.COMPARATOR_BY_NAME_AND_ID)
+                .collect(toList());
```

Comment:
Nevermind my comment. You are iterating it within another iteration, using the stream directly would consume it.